utilities/transactions/write_unprepared_transaction_test.cc

```diff
@@ -35,26 +35,32 @@ INSTANTIATE_TEST_CASE_P(
     ::testing::Values(std::make_tuple(false, false, WRITE_UNPREPARED),
                       std::make_tuple(false, true, WRITE_UNPREPARED)));
 
-enum StressAction { NO_SNAPSHOT, RO_SNAPSHOT, REFRESH_SNAPSHOT };
-class WriteUnpreparedStressTest : public WriteUnpreparedTransactionTestBase,
-                                  virtual public ::testing::WithParamInterface<
-                                      std::tuple<bool, StressAction>> {
+enum SnapshotAction { NO_SNAPSHOT, RO_SNAPSHOT, REFRESH_SNAPSHOT };
```

Comment:
Does `NO_SNAPSHOT` and `REFRESH_SNAPSHOT` covered by stress test equivalent? I thought now we only verify MultiGet result against ExpectedValue under `do_consistency_check=true` (which always `readoptionscopy.snapshot = db_->GetSnapshot();`)

Reply:
These enum  in the unit test was used to guide the transaction to do things in specific mode. With `NO_SNAPSHOT`, the read option doesn't take a snapshot. With `RO_SNAPSHOT`, a snapshot is taken and used once before any write is started. With `REFRESH_SNAPSHOT`, a snapshot is taken after each write. 

When it comes to verification regarding the different snapshot modes, the test cannot access the db's internal sequence number, so it has no direct way to verify the returned value actually respect the snapshot provided in read options. The unit test instead mocked this verification by associate db's snapshot action with an observable number `snapshot_num = counter.fetch_add(1);` And also associate the values being written with that same counter.

With this, for `NO_SNAPSHOT, the snapshot counter should stay as the lowest number, making all value counters bigger than the snapshot counter, for `RO_SNAPSHOT`, it checks the value counter has a number bigger than the sequence counter, verifying all writes happen after the action of taking a snapshot, for `REFRESH_SNAPSHOT`, the value counter has a number smaller than the snapshot counter, verifying all writes happen before the last action of taking a snapshot, because after each write, a snapshot is taken again.

In the stress test, we don't have ways to verify db's snapshot correctness like this, by that I mean, stress test provides a snapshot to the db read and trust that this snapshot is respected, without having its own verification that it is indeed the case. So I think there is no straightforward way to carry this snapshot correctness testing into the stress test.